.github/helpers/check-npm-dependency-versions.js

```diff
@@ -0,0 +1,170 @@
+const https = require('https');
+const path = require('path');
+
+// Get package.json file.
+const cmdLineArgument = process.argv.slice(2)[0];
+const packageJsonPath = cmdLineArgument.startsWith('./')
+  ? path.resolve(cmdLineArgument)
+  : path.resolve(`./${cmdLineArgument}`);
+const packageJson = require(packageJsonPath);
+
+// Read the package.json file and get the list of dependencies and devDependencies.
+const dependencies = Object.entries(packageJson.dependencies) ?? undefined;
+const devDependencies = Object.entries(packageJson.devDependencies) ?? undefined;
+
+// Results from running checkVersions.
+let dependencyResults = [];
+let devDependencyResults = [];
+
+// Check the latest version of each dependency.
+const checkVersions = async (dependencyList) => {
+  // For each dependency in the dependencyList.
+  for (let key in dependencyList) {
+    const [dependency, version] = dependencyList[key];
+    const url = `https://registry.npmjs.org/${dependency}/latest`;
+
+    try {
+      // Make an http request to the npm registry.
+      const data = await new Promise((resolve, reject) => {
+        https.get(url, (res) => {
+          let data = '';
+          res.on('data', (chunk) => {
+            data += chunk;
+          });
+          res.on('end', () => {
+            resolve(data);
+          });
+          res.on('error', (error) => {
+            reject(error);
+          });
+        });
+      });
+
+      // Parse response data for latest version.
+      const latestVersion = JSON.parse(data).version;
+
+      // Check if theres a difference in version and latestVersion.
+      if (
+        latestVersion &&
+        latestVersion !== '0.0.0' &&
+        latestVersion !== version &&
+        !latestVersion.includes('-')
+      ) {
+        // Create arrays of version triplets  ie. 2.7.4  >>  [2, 7, 4]
+        const versionTriplet = version.split('.');
+        const latestVersionTriplet = latestVersion.split('.');
+
+        // Determine version change.
+        let versionChange = 'Patch';
+        if (versionTriplet[0] !== latestVersionTriplet[0]) {
+          versionChange = 'Major';
+        } else if (versionTriplet[1] !== latestVersionTriplet[1]) {
+          versionChange = 'Minor';
+        }
+
+        // Save results.
+        const isDevDependencies = dependencyList === devDependencies;
```

Comment:
This will never be true because array equality is done by reference not by value.

Reply:
Added check by JSON string instead.

Reply:
I see what you mean but it does work